### PR TITLE
perftest: Initial import into Mariner

### PR DESCRIPTION
**Pull request by rlmenge**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

Add NVIDIA perftest package.

Note that generic upstream is found here: https://github.com/linux-rdma/perftest.git

From README
"This is a collection of tests written over uverbs intended for use as a
performance micro-benchmark. The tests may be used for HW or SW tuning
as well as for functional testing."

###### Change Log  

- Import NVIDIA's perftest into Mariner

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- local build

### CODE REVIEW

File: SPECS/perftest/perftest.spec [hunk 1]
```diff
@@ -0,0 +1,80 @@
+Summary:        IB Performance tests
+Name:           perftest
+# Update extended_release with version updates
+Version:        4.5
+Release:        1%{?dist}
+License:        BSD 3-Clause, GPLv2
```
Comment (oliviacrain) [inline SPECS/perftest/perftest.spec:6]: We should use `and` and `or` as separators, to make the licensing situation  unambiguous.

Comment (oliviacrain) [reply SPECS/perftest/perftest.spec:6]: Also, `BSD 3-Clause` can be shortened to `BSD`.

File: SPECS/perftest/perftest.spec [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+Vendor:         Microsoft Corporation
+Distribution:   Mariner
+Group:          Productivity/Networking/Diagnostic
+URL:            https://www.openfabrics.org
+Source0:        https://linux.mellanox.com/public/repo/doca/1.3.0/extras/mlnx_ofed/5.6-1.0.3.3/SOURCES/perftest_4.5.orig.tar.gz#/%{name}-%{version}.tar.gz
```
Comment (oliviacrain) [inline SPECS/perftest/perftest.spec:11]: Can we replace the `4.5` in the URL with a macro?

File: SPECS/perftest/perftest.spec [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+%prep
+%autosetup -p1
+
+%build
+%configure \
+%{__make}
+chmod -x runme
+
+%install
+make DESTDIR=%{buildroot} install
```
Comment (oliviacrain) [inline SPECS/perftest/perftest.spec:30]: Nit: Could be `%make_install`

File: SPECS/perftest/perftest.spec [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+%description
+gen3 uverbs microbenchmarks release: %extended_release
+
+%prep
+%autosetup -p1
+
+%build
+%configure \
+%{__make}
```
Comment (oliviacrain) [inline SPECS/perftest/perftest.spec:26]: This looks kinda wack to me, so just making sure- the configure script wants to be passed the path to make?

Comment (rlmenge) [reply SPECS/perftest/perftest.spec:26]: Yup this is a typo which didn't raise a failure. Fixing and testing

File: SPECS/perftest/perftest.spec (hunk 2 above)
Comment (PawelWMS) [reply SPECS/perftest/perftest.spec:11]: We could do that for `perftest` as well, I believe.

Comment (PawelWMS) [inline SPECS/perftest/perftest.spec:11]: Considering the `5.6-1.0.3.3` part, should this package somehow always follow the OFED scripts one in terms of versions? If yes, we should at least add a comment.

**Review by oliviacrain — APPROVED** [review]

**Pull request merged**